src/test/ast_program_test.cpp

```diff
@@ -166,5 +181,130 @@ TESTASTCLONEANDEQUAL(RelationCopies,
 
             )");
 
+/** test removeClause, appendRelation and removeRelation */
+TEST(AstProgram, RemoveNilConstant) {
```

Comment:
Why does it matter what type the head argument is for these clauses? I don't think it factors into clause removal, so having only one of these clause-removal tests seems to be useful.